### ISSUE TITLE
-Wmaybe-uninitialized

### ISSUE THREAD

**Issue by bakerstu** [DuffsDevice/tiny-utf8#29]
I am using your library, and have a forked version here:

https://github.com/bakerstu/tiny-utf8

I use g++ and my project, as a rule, uses -wall -werror.  Therefore, I have a fix in a branch noexcept that adds pragma statements to disable -Wmaybe-uninitialized.  The fix would look something like this in tinyutf8.h.

```
<include guard>
<includes>

#if defined(__GNUG__)
#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
#pragma push
#endif

< rest of file>

#if defined(__GNUG__)
#pragma pop
#endif

<end include guard>
```

What do you think?  Is this a change you would be willing to accept as an isolated change on a new branch as a pull request?

**Comment by DuffsDevice** [comment]
Hi Stuart,

Thank you for your message!
I'd be quite willing to accept such a change and would welcome a pull request 😃
However, I'd like to override the warning on a per-case basis. How many times does the warning occour? If it is less than let's say 3 times, I'd indeed prefer a per-case fix, with a small explanation of why the variable must have been initialized at each very point.
What do you think?

Best Regards,
Jakob

**Comment by bakerstu** [comment]
Yes, your suggested approach seems reasonable, and much more safe long term.  I'll work on a pull request.

**Comment by DuffsDevice** [comment]
Cool, thanks a lot!

**Comment by bakerstu** [comment]
@DuffsDevice I spent a little bit of time on this, and I realized that fixing this with decreased scope, as you have suggested, started to get quite messy. I decided instead to simply disable forcing -Wmaybe-uninitialized to an error, and revert it back to only a warning, and only for the scope of a single header file.

Pull request #31.

**Comment by DuffsDevice** [comment]
Dear Stuart,

thanks for your efforts at exploring this direction. When I wanted to apply your solution, I realized, that `GCC diagnostic warning` will add the warning even if "-Wno-maybe-uninitialized" is being passed through the command line, kinda pittyful.
I guess, you convinced me, that the first solution you suggested in this issue is the right 😜.

Thanks a lot!
Jakob

Closed by Merging #31 in Commit 58271596168ec6ab1a5288c81947f9a8729005b6
